### PR TITLE
com_fields vs cli finder indexer

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue  #24607

### Summary of Changes
always add table search path

### Testing Instructions
run finder indexr from cli no warning expected

### Expected result
custom fields works as before
and cli finder_indxer runs without warning

### Actual result
cli finder_indexer gives warning

cc:@laoneo

### CODE REVIEW

File: administrator/components/com_fields/models/group.php [hunk 1]
```diff
@@ -71,10 +71,8 @@ public function save($data)
 	 */
 	public function getTable($name = 'Group', $prefix = 'FieldsTable', $options = array())
 	{
-		if (strpos(JPATH_COMPONENT, 'com_fields') === false)
-		{
-			$this->addTablePath(JPATH_ADMINISTRATOR . '/components/com_fields/tables');
-		}
+
```
Comment (Quy) [inline administrator/components/com_fields/models/group.php]:
```suggestion
```

**Comment by HLeithner** [comment]
@jjnxpct can you test this pr?

File: administrator/components/com_fields/models/group.php [hunk 2]
```diff
@@ -71,10 +71,7 @@ public function save($data)
 	 */
 	public function getTable($name = 'Group', $prefix = 'FieldsTable', $options = array())
 	{
-		if (strpos(JPATH_COMPONENT, 'com_fields') === false)
-		{
-			$this->addTablePath(JPATH_ADMINISTRATOR . '/components/com_fields/tables');
-		}
+		$this->addTablePath(JPATH_ADMINISTRATOR . '/components/com_fields/tables');
```
Comment (laoneo) [inline administrator/components/com_fields/models/group.php:74]: I would add here a comment why this is necessary as normally you don't have to write this in a model. The environment should be set up properly upfront.

Comment (alikon) [reply administrator/components/com_fields/models/group.php:74]: or did you think it is better to add a define with `JPATH_COMPONENT`  in the cli script ?

**Comment by jjnxpct** [comment]
I tested this and the error is now gone in the cron email. Excellent.

**Comment by ghost** [comment]
@jjnxpct please mark your test as successfully at Issue Tracker.

File: administrator/components/com_fields/models/group.php (hunk 2 above)
Comment (laoneo) [reply administrator/components/com_fields/models/group.php:74]: What calls the model? I guess from [here](https://github.com/joomla/joomla-cms/blob/staging/plugins/system/fields/fields.php#L508)  and then is the helper [here](https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_fields/helpers/fields.php#L94) loading the items. I would probably do it in the helper.

**Comment by jjnxpct** [comment]
> @jjnxpct please mark your test as successfully at Issue Tracker.

@franz-wohlkoenig : Not sure how to do that. Sorry, I'm getting lost in this GitHub stuff...

**Comment by ghost** [comment]
No Problem @jjnxpct. At https://docs.joomla.org/Testing_Joomla!_patches scroll to Subhead "Recording test results", this might be enough information. If not, please comment :-)

**Comment by jjnxpct** [comment]
I have tested this item :white_check_mark: successfully on 4506625b478b88bcb7abb745d25479c031ec02ad<br /><br />Tested this on our live site where the problem occurred. After the code change the errors were gone in the cron output mail.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24630">issues.joomla.org/tracker/joomla-cms/24630</a>.</sub>

**Comment by jjnxpct** [comment]
@franz-wohlkoenig : Thanks! Just added the successful test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24630">issues.joomla.org/tracker/joomla-cms/24630</a>.</sub>

**Comment by ghost** [comment]
@ReLater can you please test?

**Comment by jjnxpct** [comment]
Hi! After updating Joomla the errors came back, because we tested this by changing a core file and now this file has been overriden by the Jomla update.

So I guess this means this has not been added to the Joomla release. Not sure how this works, but when can this 'fix' be expected in a next Joomla release? So we don't have to keep patching this ;-)

**Comment by HLeithner** [comment]
We need 2 successful tests then it can be merged in to core.

**Comment by jjnxpct** [comment]
OK, thanks for letting me know. Hope someone else can test this to.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4e4423ed2dd05a0b5d56c6d6f68d38369802dd46<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24630">issues.joomla.org/tracker/joomla-cms/24630</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24630">issues.joomla.org/tracker/joomla-cms/24630</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Pull request merged**